### PR TITLE
[16.0][ADD] shopinvader_api_lead: FastAPI service for creating CRM leads

### PR DESCRIPTION
**Pull request by marielejeune**
Adding a new `lead_router` and a new service to create CRM leads.
This addon is now very simple, as it just adds few info: email, subject, description. We don't need anything more for our customer, but this addon can evolve following the community's needs.

### CODE REVIEW

**Review by benwillig — APPROVED** [review]

**Comment by sebastienbeau** [comment]
There is already a PR for the lead here : https://github.com/shopinvader/odoo-shopinvader/pull/1405/

The main difference is the fields available.

On the previous version more field where available : https://github.com/shopinvader/odoo-shopinvader/blob/0b031e87c4a64e7f013598e2f8c76ac206c126cd/shopinvader_lead/services/lead.py#L33

On the other PR the list is here : https://github.com/shopinvader/odoo-shopinvader/pull/1405/files#diff-4384e51893bb83e7bdf819ff2e8d1c698a694dcf88df6e075c05d0eb05537cfdR4

File: shopinvader_api_lead/schemas/lead.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from extendable_pydantic import StrictExtendableBaseModel
+
+
+class Lead(StrictExtendableBaseModel):
+    id: int
```
Comment (sebastienbeau) [inline shopinvader_api_lead/schemas/lead.py:9]: Add the missing fields

**Review by sebastienbeau — CHANGES_REQUESTED** [review]

**Review by lmignon — APPROVED** [review]

File: shopinvader_api_lead/schemas/lead.py (hunk 1 above)
Comment (lmignon) [reply shopinvader_api_lead/schemas/lead.py:9]: I propose to make a new PR to add these missing fields

Comment (sebastienbeau) [reply shopinvader_api_lead/schemas/lead.py:9]: Ok for me

**Review by sebastienbeau — APPROVED** [review]

**Pull request merged**